src/fmu/tools/rms/zone_mapping.py

```diff
@@ -0,0 +1,242 @@
+from typing import Any
+
+DEBUG_OFF = 0
+DEBUG_ON = 1
+DEBUG_VERBOSE = 2
+DEBUG_VERY_VERBOSE = 3
+
+
+class ZoneMapping:
+    """
+    Keep info about available zones in modelling grid,
+    its zone numbers in the zone parameter and corresponding zone names.
+    The grid object contains info about zone names for the grid and
+    number of layers per zone as the most important info in this class.
+    The zone parameter (3D parameter with zone number per grid cell) contains
+    a table with corresponding zone name and zone number.
+    """
+
+    grid_model: Any = None
+    grid: Any = None
+    real_number: int = 0
+
+    # Key is zone index in the zones in the grid
+    grid_zone_dict: dict[int, dict] = {}
+
+    # Key is zone_number and value is zone name
+    zone_code_names: dict[int, str] = {}
+
+    # Key is zone_number
+    grid_zone_index_dict: dict[int, int] = {}
+
+    def __init__(
+        self,
+        grid_model: Any,
+        grid: Any,
+        real_number: int = 0,
+        fmu_mode: bool = False,
+        debug_level: int = DEBUG_OFF,
+    ):
```

Comment:
It appears that you first make variables as class atrributes, and you overwrite with instance attributes (.e.g `grid_model` is first a class attribute, and then becomes an instance attribite after `__init__`). I propose you remove the class attributes unless you have a spesific use for them?